src/commcare_cloud/environment/secrets/backends/ansible_vault/main.py

```diff
@@ -134,7 +134,12 @@ def _get_secret(self, var_name):
         return context[var_name]
 
     def _set_secret(self, var, value):
-        raise NotImplementedError
```

Comment:
Might be worth leaving a comment here to the same effect as what you wrote in the description, that no attempt was made to preserve yaml formatting or comments. Also that no edge-cases (creating a new file if it doesn't exist, etc.) are handled.